qiskit_experiments/library/tomography/basis/local_basis.py

```diff
@@ -0,0 +1,414 @@
+            self._size = len(instructions)
+            if default_states is None:
+                default_states = self._instructions
+                self._custom_defaults = False
+
+        # Construct default states
+        if default_states is not None:
+            self._default_states = tuple(DensityMatrix(i).data for i in default_states)
+            self._default_dim = self._default_states[0].shape[0]
+            if self._size is None:
+                self._size = len(self._default_states)
+            elif len(self._default_states) != self._size:
+                raise QiskitError(
+                    "Number of instructions and number of default states must be equal."
+                )
+
+        # Construct states of specific qubits if provided
+        qubit_states = qubit_states or {}
+        for qubit, states in qubit_states.items():
+            if self._size is None:
+                self._size = len(states)
+            elif len(states) != self._size:
+                raise QiskitError("Number of instructions and number of states must be equal.")
+
+            qstates = tuple(DensityMatrix(i).data for i in states)
+            self._qubit_states[qubit] = qstates
+            self._qubit_dim[qubit] = qstates[0].shape[0]
+            self._qubits.add(qubit)
+
+        # Pseudo hash value to make basis hashable for LRU cached functions
+        self._hash = hash(
+            (
+                type(self),
+                self._name,
+                self._size,
+                self._default_dim,
+                self._custom_defaults,
+                tuple(self._qubits),
+                tuple(self._qubit_dim.values()),
+                (type(i) for i in self._instructions),
+            )
+        )
+
+    def __repr__(self):
+        return f"<{type(self).__name__}: {self.name}>"
+
+    def __hash__(self):
+        return self._hash
+
+    def __eq__(self, value):
+        return (
+            super().__eq__(value)
+            and self._size == getattr(value, "_size", None)
+            and self._default_dim == getattr(value, "_default_dim", None)
+            and self._custom_defaults == getattr(value, "_custom_defaults", None)
+            and self._qubits == getattr(value, "_qubits", None)
+            and self._qubit_dim == getattr(value, "_qubit_dim", None)
+            and self._instructions == getattr(value, "_instructions", None)
+        )
+
+    def index_shape(self, qubits: Sequence[int]) -> Tuple[int, ...]:
+        return len(qubits) * (self._size,)
+
+    def matrix_shape(self, qubits: Sequence[int]) -> Tuple[int, ...]:
+        return tuple(self._qubit_dim.get(i, self._default_dim) for i in qubits)
+
+    def circuit(self, index: Sequence[int], qubits: Sequence[int]) -> QuantumCircuit:
+        # pylint: disable = unused-argument
+        if not self._instructions:
+            raise NotImplementedError("This basis does not define circuits")
```

Comment:
Here the exception is not caught. Likely we cannot define experiment without instructions (otherwise we need to override circuit method for new basis). Perhaps we can construct circuits from operator rather than raising an error here, i.e. append it a unitary gate and then let the transpiler synthesize the basis gates.
https://github.com/Qiskit/qiskit-experiments/blob/0f0684f3ef839d1568d5fc68f216254d3c762ea2/qiskit_experiments/library/tomography/tomography_experiment.py#L168

Reply:
You can define a basis without instructions if you are just using it for analysis (fitter). Circuits are only needed for the Experiment for circuit generation. So if you want to try out different states/POVMS for fitting for error mitigation or something on data youve already generated, you only need to put in them, not the instructions.

Reply:
I see. I was not able to realize this use case from the implementation. Can you catch the error in the circuit factory and say something like `f"Given basis {prep_element.name} can be used only for analysis."`?